### ISSUE TITLE
Too strict `mypy` dependency

### ISSUE THREAD

**Issue by bhuvansingla** [Portkey-AI/portkey-python-sdk#76]
### What happened?

My application uses `mypy = ">=1.2.0"`. The Portkey SDK setup requires `mypy==0.991`.
https://github.com/Portkey-AI/portkey-python-sdk/blob/366066db17c0202615a3ab2db2ad0bde6b3c9e73/setup.cfg#L23
It would be great if this constraint could be relaxed! This is a case similar to https://github.com/Portkey-AI/portkey-python-sdk/issues/66.

### Version

0.1.xx (Default)

### Relevant log output

```shell
Because portkey-ai (1.1.4) depends on mypy (0.991)
 and no versions of portkey-ai match >1.1.4,<2.0.0, portkey-ai (>=1.1.4,<2.0.0) requires mypy (0.991).
So, because <my-app> depends on both portkey-ai (^1.1.4) and mypy (>=1.2.0), version solving failed.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bhuvansingla** [comment]
I have also opened a pull request here: https://github.com/Portkey-AI/portkey-python-sdk/pull/77

**Comment by VisargD** [comment]
This is now available in v1.1.5
